pkg/platform/openstack/kuryr_bootstrap.go

```diff
@@ -189,6 +194,55 @@ func getSavedAnnotation(kubeClient client.Client, annotation string) (string, er
 	return cm.Annotations[annotation], nil
 }
 
+func getMasterMachines(kubeClient client.Client) (*machineapi.MachineList, error) {
```

Comment:
Probably MachineSets could be used instead of Machine,` oc get MachineSets -n openshift-machine-api`, this is created since the first steps of installation and also has a label applied `machine.openshift.io/cluster-api-machine-role: worker`.
As the subnet is only needed for mtu and router discovery, it should be fine to rely on the master subnet as the mtu and router would need to be the same, so no strong opinions about this.

Reply:
actually I see there are sg rules that uses the worker cidr, this could end up not being the same for master subnets

Reply:
The master machines have no MachineSet, moreover it is totally valid to create Machines without MachineSets (just like Pods without any higher-level controllers like DaemonSet). So I'd rather stick to using lowest level resources - Machines.

Reply:
Ah, and regarding the SGs - the assumption is that if somebody spawns nodes on a separate subnet, then he or she is responsible for creating all the SG rules. Martin prepared a playbook for the bare SoS and I have one for Kuryr too.

Reply:
okay